e2e/web/specs/hints.spec.ts

```diff
@@ -0,0 +1,1190 @@
+import {
+  AMEND_NOTES_CORRECT_AND_INCORRECT_CELL_GAME,
+  AMEND_NOTES_CORRECT_CELL_GAME,
+  AMEND_NOTES_EMPTY_CELL_GAME,
+  AMEND_NOTES_INCORRECT_CELL_GAME,
+  HIDDEN_PAIR_BOX_GAME,
+  HIDDEN_PAIR_COLUMN_GAME,
+  HIDDEN_PAIR_PROFILE,
+  HIDDEN_PAIR_ROW_GAME,
+  HIDDEN_QUADRUPLET_BOX_GAME,
+  HIDDEN_QUADRUPLET_COLUMN_GAME,
+  HIDDEN_QUADRUPLET_PROFILE,
+  HIDDEN_QUADRUPLET_ROW_GAME,
+  HIDDEN_SINGLE_BOX_GAME,
+  HIDDEN_SINGLE_COLUMN_GAME,
+  HIDDEN_SINGLE_PROFILE,
+  HIDDEN_SINGLE_ROW_GAME,
+  HIDDEN_TRIPLET_BOX_GAME,
+  HIDDEN_TRIPLET_COLUMN_GAME,
+  HIDDEN_TRIPLET_PROFILE,
+  HIDDEN_TRIPLET_ROW_GAME,
+  NAKED_PAIR_BOX_GAME,
+  NAKED_PAIR_COLUMN_GAME,
+  NAKED_PAIR_PROFILE,
+  NAKED_PAIR_ROW_GAME,
+  NAKED_QUADRUPLET_BOX_GAME,
+  NAKED_QUADRUPLET_COLUMN_GAME,
+  NAKED_QUADRUPLET_PROFILE,
+  NAKED_QUADRUPLET_ROW_GAME,
+  NAKED_SINGLE_GAME,
+  NAKED_TRIPLET_BOX_GAME,
+  NAKED_TRIPLET_COLUMN_GAME,
+  NAKED_TRIPLET_PROFILE,
+  NAKED_TRIPLET_ROW_GAME,
+  POINTING_PAIR_COLUMN_GAME,
+  POINTING_PAIR_PROFILE,
+  POINTING_PAIR_ROW_GAME,
+  POINTING_TRIPLET_COLUMN_GAME,
+  POINTING_TRIPLET_PROFILE,
+  POINTING_TRIPLET_ROW_GAME,
+  SIMPLIFY_NOTES_BOX_GAME,
+  SIMPLIFY_NOTES_COLUMN_GAME,
+  SIMPLIFY_NOTES_ROW_GAME,
+} from "../data";
+import { SudokuBoardComponent } from "../components/sudoku-board.component";
+import { test } from "../fixture";
+import {
+  NOT_HIGHLIGHTED_COLOR_RGB,
+  SELECTED_COLOR_RGB,
+} from "../../../app/Styling/HighlightColors";
+
+test.describe("hint mode operates correctly", () => {
+  test.use({ gameToResume: AMEND_NOTES_EMPTY_CELL_GAME });
+
+  test("selected cells are unselected when entering hint mode", async ({
+    resumeGame,
+  }) => {
+    const sudokuBoard = new SudokuBoardComponent(resumeGame);
+    await sudokuBoard.cell[0][0].click();
+    await sudokuBoard.cellHasColor(0, 0, SELECTED_COLOR_RGB);
+    await sudokuBoard.hint.click();
+    await sudokuBoard.cellHasColor(0, 0, NOT_HIGHLIGHTED_COLOR_RGB);
+  });
+
+  test("cells cannot be selected when in hint mode", async ({ resumeGame }) => {
+    const sudokuBoard = new SudokuBoardComponent(resumeGame);
+    await sudokuBoard.cellIsEnabled(0, 0);
+    await sudokuBoard.hint.click();
+    await sudokuBoard.cellIsDisabled(0, 0);
+  });
+});
+
+test.describe("board AMEND_NOTES", () => {
+  test.use({ gameToResume: AMEND_NOTES_EMPTY_CELL_GAME });
+
+  test("with empty cell and 3 groups", async ({ resumeGame }) => {
+    const notHighlightedColor = (row: number, column: number) => {
+      return (row <= 2 && column <= 2) || row === 0 || column === 0;
+    };
+
+    const hintSelectedColor = (row: number, column: number) => {
+      return (
+        (row === 0 && column === 2) ||
+        (row === 0 && column === 4) ||
+        (row === 0 && column === 7) ||
+        (row === 1 && column === 2) ||
+        (row === 2 && column === 1) ||
+        (row === 2 && column === 2) ||
+        (row === 4 && column === 0) ||
+        (row === 6 && column === 0)
+      );
+    };
```

Comment:
I don't think it is worth it to do this refactor, since sometimes we aren't specifying cells but entire rows, columns, and/or boxes. And it would be inconvenient to list every cell from a row, column, or box.